typing/jkind.ml

```diff
@@ -2335,6 +2319,22 @@ let for_unboxed_record lbls =
   in
   Builtin.product ~why:Unboxed_record tys_modalities layouts
 
+let for_non_float ~(why : History.value_creation_reason) =
+  fresh_jkind
+    { layout = Sort (Base Value);
+      mod_bounds =
+        Mod_bounds.create ~locality:Locality.Const.max
+          ~linearity:Linearity.Const.max ~portability:Portability.Const.max
+          ~yielding:Yielding.Const.max ~uniqueness:Uniqueness.Const_op.max
+          ~contention:Contention.Const_op.max
+          ~statefulness:Statefulness.Const.max
+          ~visibility:Visibility.Const_op.max ~externality:Externality.max
+          ~nullability:Nullability.Non_null ~separability:Separability.Non_float;
+      with_bounds = No_with_bounds
+    }
+    ~annotation:(mk_annot "non_float_value")
```

Comment:
I think we don't use this annotation field much (at all?), but it should probably be `value mod non_float`.

Reply:
Oops. I think it actually should be `None`, since this jkind doesn't come from annotations. Fixed.